frame/support/src/traits/hooks.rs

```diff
@@ -227,10 +227,16 @@ pub trait Hooks<BlockNumber> {
 	fn on_finalize(_n: BlockNumber) {}
 
 	/// This will be run when the block is being finalized (before `on_finalize`).
-	/// Implement to have something happen using the remaining weight.
-	/// Will not fire if the remaining weight is 0.
-	/// Return the weight used, the hook will subtract it from current weight used
-	/// and pass the result to the next `on_idle` hook if it exists.
+	///
+	/// Implement to have something happen using the remaining weight. Will not fire if the
+	/// remaining weight is 0.
+	///
+	/// The order of the pallets in the runtime enum does not affect the order in which each
+	/// pallet's `on_idle` hook is called. Instead, `on_idle` hooks are called in a round-robin
+	/// fashion indexed by the block number.
```

Comment:
Technically, the order of the pallets in the runtime does affect which pallet's on_idle is called. The starting pallet `on_idle` per block is round robin based on the order of pallets in the runtime enum, and if the weight is remaining, pallets are executed in the order of runtime as well. 😁 
```suggestion
/// The order of the pallets in the runtime enum determines the order in which each
/// pallet's `on_idle` hook is called with each pallet `on_idle` hook getting chosen to be the
/// first to execute in a round robin fashion indexed by the block number.
```